File: whatRubyCanLearnFromJava/mocks.java
public class MocksTest {

    @Test
    public void shouldBeAbleToDrive() {
      //arrange
      Car car = mock(Car.class);
      Person p = new Person(car);

      //act
      p.goSomeplace():

      //assert
      verify(car).drive();
    }


}

File: whatRubyCanLearnFromJava/mocks.rb
require 'spec'

    describe "person" do
      it "should be able to drive a car" do
        #arrange
        car = mock()
        car.should_receive(:drive)
        person = Person.new(:car => car)

        #act
        person.go_someplace

        #where's my assert?
      end
    end


    describe "person" do
      it "should be able to drive a car" do
        #arrange
        car = Car.new
        stub(car).drive
        person = Person.new(:car => car)

        #act
        person.go_someplace

        #assert
        car.should have_received.drive
      end
    end
    
